components/dex.libsonnet

```diff
@@ -109,6 +133,9 @@
     apiVersion: 'v1',
     kind: 'Service',
     metadata: {
+      annotations: {
+        'service.beta.openshift.io/serving-cert-secret-name': dex.config.tlsSecret,
```

Comment:
should this go in the upstream config? Or rather in openshift-specific down-stream configuration?

cc @metalmatze

Reply:
This has no side-effect for non-openshift deployments. We have a similar case in the Jaeger Operator where we add Istio and Linkerd annotations to some deployments, even though we might not have any in the cluster (and never both at the same time). The extra complexity isn't worth it, IMO.

Reply:
Just noticed you mentioned a "downstream configuration". Is it in this repo? It might indeed be easy enough to define this annotation there.

Reply:
exactly, it's not in this repo, but it is also jsonnet and it inherits this config. We have all of our openshift stuff there, so let's be consistent and keep this repo as agnostic as possible to openshift.